rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/uri/FallbackResolver.java

```diff
@@ -250,5 +272,45 @@ public void unwatch(ISourceLocation root, Consumer<ISourceLocationChanged> watch
         getClient().removeWatcher(root, watcher, getServer());
 
     }
+    
+    public boolean isFileManaged(ISourceLocation file) {
+        for (final var service : textDocumentServices) {
+            if (service.isManagingFile(file)) {
+                return true;
+            }
+        }
+        return false;
+    }
 
+    @Override
+    public ISourceLocation resolve(ISourceLocation input) throws IOException {
+        if (isFileManaged(input)) {
+            try {
+                return URIUtil.changeScheme(input, "lsp+" + input.getScheme()).top();
+            } catch (URISyntaxException e) {
+                // fall through
+            }
+        }
+        return input;
+    }
+
+    @Override
+    public String authority() {
+        throw new UnsupportedOperationException("'authority' not supported by fallback resolver");
+    }
+
+    private Set<IBaseTextDocumentService> textDocumentServices = new HashSet<>();
```

Comment:
nit: should be final

notnit: not a threadsafe container, since we're only iterating over it, never clearing any, or using the key of it, an `CopyOnWriteArrayList` might be the better option.